addons/addon-base-raas/packages/base-raas-services/lib/data-egress/data-egress-service.js

```diff
@@ -263,6 +259,156 @@ class DataEgressService extends Service {
+        Body: JSON.stringify({ objects: objectList }),
+        ContentType: 'application/json',
+      };
+      await s3Service.putObject(params);
+    } catch (error) {
+      throw this.boom.badRequest(
+        `Error in preparing EgressStoreSnapshot, bucket:${egressNotificationBucketName}, key: ${key}`,
+        true,
+      );
+    }
+    return { bucket: egressNotificationBucketName, key };
+  }
+
+  async notifySNS(requestContext, environmentId) {
+    const enableEgressStore = this.settings.get(settingKeys.enableEgressStore);
+    const curUser = _.get(requestContext, 'principalIdentifier.uid');
+    if (!enableEgressStore || enableEgressStore.toUpperCase() !== 'TRUE') {
+      throw this.boom.forbidden('Unable to create Egress store since this feature is disabled', true);
+    }
+
+    const egressStoreInfo = await this.getEgressStoreInfo(environmentId);
+    if (!egressStoreInfo.isAbleToSubmitEgressRequest) {
+      throw this.boom.badRequest(
+        `Egress Store:${egressStoreInfo.id} is not ready for egress. Please contact your administrator for more information.`,
+        true,
+      );
+    }
+    const isEgressStoreOwner = egressStoreInfo.createdBy === curUser;
+    if (!isAdmin(requestContext) && !isEgressStoreOwner) {
+      throw this.boom.forbidden(
+        `You are not authorized to submit egress request. Please contact your administrator for more information.`,
+        true,
+      );
+    }
+    const egressStoreObjectList = await this.prepareEgressStoreSnapshot(egressStoreInfo);
+
+    // update dynamodb info
+    const egressStoreDdbLockId = `egress-store-ddb-access-${egressStoreInfo.id}`;
+    if (egressStoreInfo.status.toUpperCase() !== PENDING_STATUS_CODE) {
+      egressStoreInfo.status = PENDING_STATUS_CODE;
+    }
+    egressStoreInfo.updatedBy = curUser;
+    egressStoreInfo.updatedAt = new Date().toISOString();
+    egressStoreInfo.isAbleToSubmitEgressRequest = false;
+    egressStoreInfo.egressStoreObjectListLocation = `arn:aws:s3:::${egressStoreObjectList.bucket}/${egressStoreObjectList.key}`;
+    const curVersion = parseInt(egressStoreInfo.ver, 10);
+    egressStoreInfo.ver = curVersion + 1;
+    await this.lockAndUpdate(egressStoreDdbLockId, egressStoreInfo.id, egressStoreInfo);
+
+    const message = {
+      egressStoreObjectListLocation: `arn:aws:s3:::${egressStoreObjectList.bucket}/${egressStoreObjectList.key}`,
+      id: uuid.v4(),
+      egress_store_id: egressStoreInfo.id,
+      egress_store_name: egressStoreInfo.egressStoreName,
+      created_at: egressStoreInfo.createdAt,
+      created_by: egressStoreInfo.createdBy,
+      workspace_id: egressStoreInfo.workspaceId,
+      project_id: egressStoreInfo.projectId,
+      s3_bucketname: egressStoreInfo.s3BucketName,
+      s3_bucketpath: egressStoreInfo.s3BucketPath,
+      status: egressStoreInfo.status,
+      updated_by: egressStoreInfo.updatedBy,
+      updated_at: egressStoreInfo.updatedAt,
+      ver: egressStoreInfo.ver,
```

Comment:
Find it odd that this version will be plus one the version in `egressStoreObjectListLocation`.  Shouldn't they be same?

Reply:
The version needs to be updated here since the notification method is triggered and we need to take it as a new snapshot of egress store so that objects in egress store metadata is stored in a new S3 location and DDB of egress store info is updated.

Reply:
fixed in commit: 18639d639b979c9054fa5d24fa94b3a0bdd98f11